### PR TITLE
feat: Extend the How to use guide in the README

### PR DESCRIPTION
**Pull request by denessapi**
Add an option to use the app without using or extending the TrapApplication class.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,9 +14,15 @@ This library can collect various device and user data, forwarding it to a specif
...
 ## How to use it
-You can check out the Example app in this project for a working example. The application needs to either use the provided TrapApplication as the 'android:name' value for the '<application>' tag in the AndroidManifest.xml, or subclass TrapApplication if you also need to use a custom Application class. Configuration is done via a '<meta-data>' tag within the '<application>' tag.
 
-### 1. Specify the application class
+### Option A - Using a custom Application implementation
+
+You can check out the Example app in this project for a working example. The application needs to
+either use the provided TrapApplication as the `android:name` value for the `<application>` tag in
+the AndroidManifest.xml, or subclass TrapApplication if you also need to use a custom Application
+class. Configuration is done via a `<meta-data>` tag within the `<application>` tag.
```
Comment (gbence) [inline README.md]: Please enclose file names and class names with backticks.

File: README.md [hunk 2]
```diff
@@ -14,9 +14,15 @@ This library can collect various device and user data, forwarding it to a specif
...
+
+#### 1. Specify the application class
 ```xml
```
Comment (gbence) [inline README.md:27]: Please add a newline here.

File: README.md [hunk 3]
```diff
@@ -37,7 +43,7 @@ You can check out the Example app in this project for a working example. The app
 </manifest>
 ```
 
-### 2. (Optional) Use a custom config class
+#### 2. (Optional) Use a custom config class
 Create the configuration provider implementation.
```
Comment (gbence) [inline README.md:49]: Newline

File: README.md [hunk 4]
```diff
@@ -71,7 +77,7 @@ class MyConfig: TrapConfigProvider {
 }
 ```
 
-Specify your configuration provider in the 'AndroidManifest.xml'.
+Specify your configuration provider in the `AndroidManifest.xml`.
 ```xml
```
Comment (gbence) [inline README.md:84]: Newline

File: README.md [hunk 5]
```diff
@@ -94,7 +100,39 @@ Specify your configuration provider in the 'AndroidManifest.xml'.
...
+    file.writeText(UUID.randomUUID().toString())
+}
+config.reporter.sessionId = UUID.fromString(file.readText())
+
+// Instantiate the TrapManager
+trapManager = TrapManager.getInstance(application, config)
+```
+
+### Ask for interactive permissions (if not previously requested) - in both cases
 ```kotlin
```
Comment (gbence) [inline README.md:140]: Newline

**Review by gbence — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (denessapi) [reply README.md:27]: Done

File: README.md (hunk 3 above)
Comment (denessapi) [reply README.md:49]: Done

File: README.md (hunk 4 above)
Comment (denessapi) [reply README.md:84]: Done

File: README.md (hunk 5 above)
Comment (denessapi) [reply README.md:140]: Done

File: README.md (hunk 1 above)
Comment (denessapi) [reply README.md]: Done

**Review by gbence — APPROVED** [review]

**Pull request merged**